src/components/JsonViewer/JsonViewer.tsx

```diff
@@ -0,0 +1,337 @@
+        }
+        dataTable.current?.scrollTo(matchedRows[index] - 6);
+        searchRef.current?.focus();
+    };
+
+    const onPrevMatch = () => {
+        onNextMatch(null, -1);
+    };
+
+    const onEnterKeyDown = (e: React.KeyboardEvent) => {
+        if (e.key !== 'Enter') {
+            return;
+        }
+        if (e.shiftKey || e.ctrlKey) {
+            onPrevMatch();
+        } else {
+            onNextMatch(null);
+        }
+    };
+
+    const renderToolbar = () => {
+        return (
+            <Flex gap={2} wrap="nowrap" className={block('toolbar')}>
+                <Flex gap={1} wrap="nowrap">
+                    <ActionTooltip title={i18n('action_expand-all')}>
+                        <Button onClick={onExpandAll} view="flat-secondary">
+                            <Icon data={ArrowDownToLineIcon} />
+                        </Button>
+                    </ActionTooltip>
+                    <ActionTooltip title={i18n('action_collapse-all')}>
+                        <Button onClick={onCollapseAll} view="flat-secondary">
+                            <Icon data={ArrowUpFromLineIcon} />
+                        </Button>
+                    </ActionTooltip>
+                </Flex>
+                {search && (
+                    <Filter
+                        onUpdate={onFilterChange}
+                        matchIndex={matchIndex}
+                        matchedRows={state.matchedRows}
+                        value={filter}
+                        ref={searchRef}
+                        onKeyDown={onEnterKeyDown}
+                        onNextMatch={onNextMatch}
+                        onPrevMatch={onPrevMatch}
+                    />
+                )}
+                <span className={block('extra-tools')}>{extraTools}</span>
+            </Flex>
+        );
+    };
+
+    const onShowFullText = (index: number) => {
+        const {
+            flattenResult: {searchIndex, data},
+        } = state;
+
+        setFullValue({
+            value: data[index].value,
+            searchInfo: searchIndex[index],
+        });
+    };
+
+    const onHideFullValue = () => {
+        setFullValue(undefined);
+    };
+
+    const renderFullValueModal = () => {
+        const {value, searchInfo} = fullValue ?? {};
+
+        const tmp = unipika.format(value, {...unipikaSettings, asHTML: false});
+
+        return (
+            value && (
+                <FullValueDialog
+                    onClose={onHideFullValue}
+                    starts={searchInfo?.valueMatch || []}
+                    text={tmp.substring(1, tmp.length - 1)}
+                    length={filter.length}
+                />
+            )
+        );
+    };
+
+    return (
+        <div className={block()}>
+            {renderToolbar()}
+            {renderTable()}
+            {renderFullValueModal()}
+        </div>
+    );
+}
+
+function asModifier(path = '') {
+    return path.replace(/[^-\w\d]/g, '_');
+}
```

Comment:
Please, provide regexp with the example: `// stringBeforeReplace => stringAfterReplace`

Reply:
It seems that this function is redundant. I removed it.